types/atom/index.d.ts

```diff
@@ -16,6 +16,10 @@ import { ChildProcess } from "child_process";
 
 declare global {
     const atom: AtomEnvironment;
+
+    interface HTMLElementTagNameMap {
+      "atom-text-editor": TextEditorElement;
+    }
```

Comment:
We're augmenting here. `HTMLElementTagNameMap` is [defined in TS's `lib.dom.d.ts`](https://github.com/Microsoft/TypeScript/blob/aa9e2ba31822ba7b002cf1c436e2c6cd6954972b/src/lib/dom.generated.d.ts#L14847) (warning: **large** file). That thing's defined in global scope, so it's impossible to augment from module scope, at least as far as I can tell.

Reply:
Oh! I didn't realize it was a TypeScript core thing, now I get the point of it being declared in global (because it *already is* in global scope.) Totally makes sense, then. Forget what I said above, I didn't catch onto the fact that it's merging the new value into an existing interface. Definitely done correctly in that case, I'll go ahead and approve as soon as the CI is done. 👍